### PR TITLE
Handle dartfmt crashes

### PR DESCRIPTION
**Pull request by jonasfj**

### CODE REVIEW

**Comment by jonasfj** [comment]
Fixes https://github.com/dart-lang/pana/issues/690

File: lib/src/sdk_env.dart [hunk 1]
```diff
@@ -500,3 +508,13 @@ String _join(String path, String binDir, String executable) {
   }
   return cmd;
 }
+
+/// Thrown when `dartfmt` fails to run.
+class DartFmtFailed implements Exception {
```
Comment (sigurdm) [inline lib/src/sdk_env.dart:513]: Let's call this `ToolException` so we can reuse it across tools?

Comment (jonasfj) [reply lib/src/sdk_env.dart:513]: Love it!

**Comment by jonasfj** [comment]
Blocked on https://github.com/dart-lang/pana/pull/691

**Pull request merged**